app/controllers/roles_controller.rb

```diff
@@ -0,0 +1,9 @@
+class RolesController < ApplicationController
+  before_action -> { nav_context(:people, :roles) }
+
+  def index
+    authorize Role
+    @roles = policy_scope(Role)
+    @roles = @roles.sort_by { |x| User::ROLES.index(x) } # this doesn't seem to work
```

Comment:
So this is a confusing case and we should have paired on this probably, but here is my thinking..

The main resource you're protecting here for auth'z is user, not role. We want to make sure we're showing people users they're not allowed to see. 

Roles currently aren't protected by any policy because they're pretty basic objects.

We also want to only show users for the current community. In a multi-community cluster, people can see all users in the cluster but on this page we only want to show the ones from the current community (just like in the directory we only show current community users.)

So I think you might consider this:

```
authorize User
@users_by_role = Hash.new { |h, k| h[k] = [] }
policy_scope(User).in_community(current_community).includes(:roles).active.by_name.each do |u| 
  u.roles.each { |r| @users_by_role[r.to_sym] << u }
end
```

Some notes:

* `includes(:roles)` does eager loading so you don't hit the roles table every time you look at a user's roles.
* `Hash.new { |h, k| h[k] = [] }` creates a hash with a default value of a new array. Try it out in the console.
* `by_name` gives us alpha sort on user names
* `active` ensures we don't display old inactive users
* `to_sym` converts to symbol, which is key because User::ROLES is an array of symbols.

Then in the view, you can iterate over each role in `ROLES` and get the matching users for each role from the hash.